packet.cpp

```diff
@@ -0,0 +1,133 @@
+#include "packet.h"
+
+Packet::Packet(char data[24], tcp::socket& socket,
+               const std::shared_ptr<KeyValueStore>& k)
+    : bod_len(readUInt32LE(data, 8)),
+      opaque(readUInt32LE(data, 12)),
+      key_len(readUInt16LE(data, 2)),
+      op(data[1]),
+      ext_len(data[4]) {
+  read(socket);
```

Comment:
Maybe some validation? Or at least some asserts, these values are blindly used elsewhere and it seems that if a malformed data stream is given, all kinds of bad things may happen.